### PR TITLE
chore(RHTAPWATCH-946): document triggering via webhooks

### PR DESCRIPTION
**Pull request by yftacherzog**
Adding documentation for triggering builds via webhooks. Adding and tweaking some other sections to support that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,27 +83,48 @@ sudo sysctl fs.inotify.max_user_instances=512
...
+The namespace where the pipeline is running should be configured with a push secret for the container registry.
 
 Tekton provides a way to inject push secrets into pipelines by attaching them to a service account.
 
-The service account used for running the pipelines is the `default` service account.
+The service account used for running the pipelines is the namespace's `default` service
+account.
 
-1. Create the secret in the namespace:
+1. Create the secret on the pipeline's namespace (see example below for extracting the
```
Comment (avi-biton) [inline README.md]:
```suggestion
1. Create the secret in the pipeline's namespace (see example below for extracting the
```

File: README.md [hunk 2]
```diff
@@ -111,6 +154,53 @@ TBA
...
+To do that, we rely on [smee](https://smee.io/): We configure a GitHub app to send
+events to a public channel we create on a public `smee` server, and we deploy a client
+within the cluster to listen to those events. The client will rely those events to
+the pipelines-as-code (Tekton) inside the cluster.
+
+1. Create a new channel in [smee](https://smee.io/), and take a note of the webhook
+   proxy URL.
+
+2. Create a GitHub app following
+   [Pipelines-as-Cod documentation](https://pipelinesascode.com/docs/install/github_apps/#manual-setup).
```
Comment (avi-biton) [inline README.md]:
```suggestion
   [Pipelines-as-Code documentation](https://pipelinesascode.com/docs/install/github_apps/#manual-setup).
```

File: README.md [hunk 3]
```diff
@@ -111,6 +154,53 @@ TBA
...
+
+   For `Homepage URL` you can insert `https://localhost:9443/` (it doesn't matter).
+
+   For `Webhook URL` insert the smee client's webhook proxy URL from previous step.
+
+   Generate and download the private key and create a secret on the cluster per the
+   instructions, providing the location of the private key, the App ID, and the openssl-generated secret created during the process.
+
+3. Install the app on the repository: Go the app page on GitHub, click on Install App on
```
Comment (avi-biton) [inline README.md]:
```suggestion
3. Install the app on the repository: Go to the app page on GitHub, click on Install App on
```

File: test/resources/demo-users/user/ns2/pull-request.yaml [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+  namespace: user-ns2
+spec:
+  params:
+  - name: dockerfile
+    value: Dockerfile
+  - name: git-url
+    value: https://github.com/yftacherzog/sample-component.git
+  - name: output-image
+    value: quay.io/yftacherzog/konflux-ci-user2:on-pr
+  - name: path-context
```
Comment (avi-biton) [inline test/resources/demo-users/user/ns2/pull-request.yaml:26]: Dont you want to use placeholders for `git-url` and `output-image`?

**Comment by avi-biton** [comment]
We need to add a step of creating the pipeline under `.tekton` folder in the application repository

File: test/resources/demo-users/user/ns2/pull-request.yaml (hunk 4 above)
Comment (yftacherzog) [reply test/resources/demo-users/user/ns2/pull-request.yaml:26]: I think these sorts of examples are less meaningful if the user cannot see the full picture, so I think we need a proper reference repository (i.e. for the repo layout, etc.). I think such repository can exist on the konflux-ci github org.

I think we can proceed with the current content and create an issue for changing the references to something not associated with personal accounts.

Make sense?

File: README.md (hunk 1 above)
Comment (yftacherzog) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (yftacherzog) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (yftacherzog) [reply README.md]: done

**Review by avi-biton — APPROVED** [review]

**Pull request merged**